src/ide/code_actions/create_module_file.rs

```diff
@@ -0,0 +1,60 @@
+use cairo_lang_defs::ids::{LanguageElementId, ModuleId};
+use cairo_lang_syntax::node::ast::{ItemModule, MaybeModuleBody};
+use cairo_lang_syntax::node::kind::SyntaxKind;
+use cairo_lang_syntax::node::{SyntaxNode, Terminal, TypedSyntaxNode};
+use lsp_types::{CodeAction, CreateFile, DocumentChangeOperation, ResourceOp, Url, WorkspaceEdit};
+
+use crate::lang::db::{AnalysisDatabase, LsSemanticGroup, LsSyntaxGroup};
+
+/// Code actions for missing module file.
+pub fn create_module_file(
+    db: &AnalysisDatabase,
+    node: SyntaxNode,
+    mut url: Url,
+) -> Option<CodeAction> {
+    let item_module = db.first_ancestor_of_kind(node.clone(), SyntaxKind::ItemModule)?;
+    let item_module = ItemModule::from_syntax_node(db, item_module);
+
+    if !matches!(item_module.body(db), MaybeModuleBody::None(_)) {
+        return None;
+    }
+
+    let mut is_crate_root = false;
+
+    if let ModuleId::Submodule(submodule) = db.find_module_containing_node(&node)? {
```

Comment:
Does this return submodule for items like `mod xyz` in `src/lib.cairo`?

Reply:
Yes